### PR TITLE
fix: display xaxis ticks with selection exceeding the data

### PR DESCRIPTION
**Pull request by QuCMGisaia**
- Fix gisaia/ARLAS-wui#411

### CODE REVIEW

File: src/histograms/charts/AbstractChart.ts [hunk 1]
```diff
@@ -843,6 +855,58 @@ export abstract class AbstractChart extends AbstractHistogram {
...
+      let fakeDataKey = +data[data.length - 1].key;
+      while (fakeDataKey < +this.selectionInterval.endvalue) {
+        fakeDataKey += +bucketSize;
+        if (this.histogramParams.dataType === DataType.numeric) {
+          extendedData.push({
+            key: fakeDataKey,
+            value: 0
+          });
+        } else {
+          extendedData.push({
```
Comment (MohamedHamouGisaia) [inline src/histograms/charts/AbstractChart.ts:899]: I have the 'intuition' that maybe utc VS notUtc might be problematic when instantiating a new Date(xxx) ?

Comment (QuCMGisaia) [reply src/histograms/charts/AbstractChart.ts:899]: After some tests, whether we use utc or not, the keys that the histogram have are still the same, the transformation happens when displaying the data ticks. The difference between buckets stay the same no matter if the histogram displays the data in utc or not. I am however going to add a safeguard regarding the number of buckets that can be plotted, as there is an intrinsic limit of 1000 if I am not mistaken.

Comment (MohamedHamouGisaia) [reply src/histograms/charts/AbstractChart.ts:908]: Ok great
For the limit, it' s 10000

File: src/histograms/charts/AbstractChart.ts [hunk 2]
```diff
@@ -66,6 +66,9 @@ export abstract class AbstractChart extends AbstractHistogram {
   protected rectangleCurrentClipper: HistogramSVGRect;
   protected selectedIntervals = new Map<string, { rect: HistogramSVGRect; startEndValues: SelectedOutputValues; }>();
 
+  /** Maximum number of buckets that a chart can have */
+  private MAX_BUCKET_NUMBER = 1000;
```
Comment (MohamedHamouGisaia) [inline src/histograms/charts/AbstractChart.ts:70]: It's 10000

**Review by MohamedHamouGisaia — APPROVED** [review]

**Pull request merged**